pensieve/tests/integration/test_analysis_integration.py

```diff
@@ -0,0 +1,154 @@
+import datetime as dt
+from google.cloud import bigquery
+from pathlib import Path
+import datetime
+import mock
+import mozanalysis
+import os
+import pytz
+import pytest
+
+from mozanalysis.metrics import Metric, DataSource, agg_sum
+from google.api_core.exceptions import NotFound
+from pensieve.analysis import Analysis, BigQueryClient
+from pensieve.experimenter import Experiment, Variant
+
+TEST_DIR = Path(__file__).parent.parent
+
+
+@pytest.mark.integration
+class TestAnalysisIntegration:
+    project_id = os.getenv("GOOGLE_PROJECT_ID")
+
+    if project_id is None:
+        print("GOOGLE_PROJECT_ID is not set.")
+        assert False
+
+    test_project = "pensieve-integration-test"
+    test_dataset = "test"
+    # contains the tables filled with test data required to run metrics analysis
+    static_dataset = "test_data"
+
+    client = bigquery.client.Client(project_id)
+
+    clients_daily_source = TEST_DIR / "data" / "test_clients_daily.ndjson"
+    events_source = TEST_DIR / "data" / "test_events.ndjson"
+    get_key_udf = TEST_DIR / "data" / "get_key.sql"
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        # remove all tables previously created
+        try:
+            self.client.delete_dataset(self.test_dataset, delete_contents=True, not_found_ok=True)
+            self.client.create_dataset(self.test_dataset)
+        except NotFound:
+            pass
+
+        try:
+            self.client.get_table(f"{self.static_dataset}.clients_daily")
+        except NotFound:
+            table_ref = self.client.create_table(f"{self.static_dataset}.clients_daily")
```

Comment:
If the project is configurable, this should handle the case where the dataset doesn't exist yet (but I think maybe the project shouldn't be configurable).